### ISSUE TITLE
Question: do proxmox nodes share auth token?

### ISSUE THREAD

**Issue by variable** [proxmoxer/proxmoxer#50]
Let's say I have setup a proxmox api load balancer that reverse proxy to my cluster.

If the load balancer switch to another node, does the existing auth token still work?

**Comment by jhollowe** [comment]
yes, if your nodes are in a cluster, load balancing across their API will work.

I tested this by creating 2 nodes (in VMs) and joining them together in a cluster. I then created a container on one node, installed nginx, and added the following to `/etc/nginx/nginx.conf`:

```
stream {
        upstream api_nodes {
                server 10.0.1.111:8006;
                server 10.0.1.112:8006;
        }

        server {
                listen 8006;
                proxy_pass api_nodes;
        }
}
```

using the container's IP address, I can access the web UI or use proxmoxer and the calls are evenly distributed between the two nodes. If I were actually doing this for continued use, I would add `hash $remote_addr;` to the upstream so connections from any give IP are tied to a specific node. This means all the traffic from your computer will go to one node, but multiple clients will be load balanced across the nodes.

I *think* load balancing individual requests between nodes is okay, but there might be weird race conditions or different responses that I haven't seen. I would try to tie connections (or at least chunks of requests) to a single node (like done above with the hash of the IP).
